src/Http/Routing/src/ValidationEndpointFilterFactory.cs

```diff
@@ -87,8 +89,27 @@ public static EndpointFilterDelegate Create(EndpointFilterFactoryContext context
             if (validateContext is { ValidationErrors.Count: > 0 })
             {
                 context.HttpContext.Response.StatusCode = StatusCodes.Status400BadRequest;
-                context.HttpContext.Response.ContentType = "application/problem+json";
-                return await ValueTask.FromResult(new HttpValidationProblemDetails(validateContext.ValidationErrors));
+
+                var problemDetails = new HttpValidationProblemDetails(validateContext.ValidationErrors);
+
+                var problemDetailsService = context.HttpContext.RequestServices.GetService<IProblemDetailsService>();
+                if (problemDetailsService is not null)
+                {
+                    if (await problemDetailsService.TryWriteAsync(new()
+                    {
+                        HttpContext = context.HttpContext,
+                        ProblemDetails = problemDetails
+                    }))
+                    {
+                        // We need to prevent further execution, because the actual
```

Comment:
I think with this pattern, we've lost the ability for users to add endpoint filters after this one that can inspect what the validation errors that were returned were. Since we lack `ModelState` in minimal APIs, this might be the only way for users to understand the error state.

Reply:
In fact, this is the point I mentioned I was unsure about.

However, I have verified that, even with the current implementation, custom endpoint filters are not invoked when validation fails, or am I missing something?

Reply:
You're correct -- I got my wires crossed. Our only support for circumventing the default filter at the moment is disabling it and adding your own explicit one. In that context, this approach seems fine.